src/phases/query/CollScanLargeProjection.yml

```diff
@@ -0,0 +1,243 @@
+    {a: {b: {c: 1, d: 1, e: 1, f: 1, g: 1, h: 1, i: 1, j: 1, k: 1, l: 1, m: 1, n: 1, o: 1, p: 1, q: 1, r: 1, s: 1, t: 1, u: 1, v: 1}}}
+
+  LargeDottedPathExclusionProjection: &LargeDottedPathExclusionProjection
+    {a: {b: {c: 0, d: 0, e: 0, f: 0, g: 0, h: 0, i: 0, j: 0, k: 0, l: 0, m: 0, n: 0, o: 0, p: 0, q: 0, r: 0, s: 0, t: 0, u: 0, v: 0}}}
+
+  LargeDottedPathRenamingProjection: &LargeDottedPathRenamingProjection
+    {a: {b: {c1: '$a.b.c', d1: '$a.b.d', e1: '$a.b.e', f1: '$a.b.f', g1: '$a.b.g', h1: '$a.b.h', i1: '$a.b.i', j1: '$a.b.j', k1: '$a.b.k', l1: '$a.b.l', m1: '$a.b.m', n1: '$a.b.n', o1: '$a.b.o', p1: '$a.b.p', q1: '$a.b.q', r1: '$a.b.r', s1: '$a.b.s', t1: '$a.b.t', u1: '$a.b.u', v1: '$a.b.v'}}}
+
+  LargeMixedProjection: &LargeMixedProjection
+    {a: {b: {c: 1, d: 1, e: 1, f: 1, g: 1, h: 1, i: 1, j: 1, k: 1, l: 1, m1: '$a.b.m', n1: '$a.b.n', o1: '$a.b.o', p1: '$a.b.p', q1: '$a.b.q', r1: '$a.b.r', s1: '$a.b.s', t1: '$a.b.t', u1: '$a.b.u', v1: '$a.b.v'}}}
+
+  LargeCommonPathPrefixInclusionProjection: &LargeCommonPathPrefixInclusionProjection
+    {a: {b: {c: {d: {e: 1, f: 1, g: 1, h: 1, i: 1, j: 1, k: 1, l: 1, m: 1, n: 1, o: 1, p: 1, q: 1, r: 1, s: 1, t: 1, u: 1, v: 1}}}}}
+ 
+  LargeCommonPathPrefixExclusionProjection: &LargeCommonPathPrefixExclusionProjection
+    {a: {b: {c: {d: {e: 0, f: 0, g: 0, h: 0, i: 0, j: 0, k: 0, l: 0, m: 0, n: 0, o: 0, p: 0, q: 0, r: 0, s: 0, t: 0, u: 0, v: 0}}}}}
+
+
+ActorTemplates:
+- TemplateName: CollScanProjectionTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "unused"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: *Repeat
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: {}
+              Options:
+                Projection: {^Parameter: {Name: "Projection", Default: "unused"}}
+
+Actors:
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 4
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document: {
+          a: {
+            b: {
+              c: {
+                d: {
+                  e: [{a: {^RandomInt: {min: 0, max: 1000}}, b: {^Inc: {start: 11, step: 2}}}],
```

Comment:
This part of leaf fields is a big dup code. Tried to dedup it with `GlobalDefults` (Please check the 3rd commit) but I failed to run the workload. Not sure if there's a better approach to dedup the code.

Reply:
I'm ok with this being duplicated for now, but if you wanted to factor it out I think you could create yet another phases yml and import it here, similar to how this [query](https://github.com/mongodb/genny/commit/091adc58b38bbe099bf1697076491e55685cccab#diff-48c7368ab151bb9b136bbac855dcca7e723fc2d481343154e9db3f33c4a96309R75) is defined in one phase yml and imported in [another](https://github.com/mongodb/genny/commit/091adc58b38bbe099bf1697076491e55685cccab#diff-90994296f65c6485553a8e1f170e1fa3330b7035a66f84c1413e280e3dd09afcR23-R26) then referenced [here](https://github.com/mongodb/genny/commit/091adc58b38bbe099bf1697076491e55685cccab#diff-90994296f65c6485553a8e1f170e1fa3330b7035a66f84c1413e280e3dd09afcR351). Might facilitate writing new queries.